pennylane/workflow/qnode.py

```diff
@@ -249,14 +255,25 @@ def _validate_gradient_kwargs(gradient_kwargs: dict) -> None:
 
 
 def _validate_qfunc_output(qfunc_output, measurements) -> None:
-    if isinstance(qfunc_output, qml.numpy.ndarray):
-        measurement_processes = tuple(measurements)
-    elif not isinstance(qfunc_output, Sequence):
-        measurement_processes = (qfunc_output,)
-    else:
-        measurement_processes = qfunc_output
-
-    if not measurement_processes or not all(
+    measurement_processes = qml.pytrees.flatten(
+        qfunc_output,
+        is_leaf=lambda obj: isinstance(obj, qml.measurements.MeasurementProcess),
+    )[0]
+
+    if len(measurement_processes) == 0:
+        measurement_processes = [qfunc_output]

```

Comment:
Which case is this for?

Reply:
For when the user either:
- Returns no measurements 
- Returns non-measurements

Reply:
Although equivalent, I think this change would make the logic more clear. 
```suggestion
    if len(measurement_processes) == 0:
        measurement_processes = [None]

```